openmc/filter.py

```diff
@@ -1610,3 +1610,222 @@ class DelayedGroupFilter(IntegralFilter):
         filter's bins.
 
     """
+
+
+class EnergyFunctionFilter(Filter):
+    """Multiplies tally scores by an arbitrary function of incident energy.
+
+    The arbitrary function is described by a piecewise linear-linear
+    interpolation of energy and y values.
+
+    Parameters
+    ----------
+    energy : Iterable of Real
+        A grid of energy values in eV.
+    y : iterable of Real
+        A grid of interpolant values in eV.
+
+    Attributes
+    ----------
+    energy : Iterable of Real
+        A grid of energy values in eV.
+    y : iterable of Real
+        A grid of interpolant values in eV.
+    num_bins : Integral
+        The number of filter bins (always 1 for this filter)
+    stride : Integral
+        The number of filter, nuclide and score bins within each of this
+        filter's bins.
+
+    """
+
+    def __init__(self, energy, y):
+        self.energy = energy
+        self.y = y
+        self._stride = None
+
+    def __eq__(self, other):
+        if type(self) is not type(other):
+            return False
+        elif not all(self.energy == other.energy):
+            return False
+        elif not all(self.y == other.y):
+            return False
+        else:
+            return True
+
+    def __gt__(self, other):
+        if type(self) is not type(other):
+            if self.short_name in _FILTER_TYPES and \
+                other.short_name in _FILTER_TYPES:
+                delta = _FILTER_TYPES.index(self.short_name) - \
+                        _FILTER_TYPES.index(other.short_name)
+                return delta > 0
+            else:
+                return False
+        else:
+            return False
+
+    def __lt__(self, other):
+        if type(self) is not type(other):
+            if self.short_name in _FILTER_TYPES and \
+                other.short_name in _FILTER_TYPES:
+                delta = _FILTER_TYPES.index(self.short_name) - \
+                        _FILTER_TYPES.index(other.short_name)
+                return delta < 0
+            else:
+                return False
+        else:
+            return False
+
+    def __hash__(self):
+        # For some reason, it seems the __hash__ method is not inherited when we
+        # overwrite __repr__.
+        return hash(repr(self))
+
+    def __repr__(self):
+        string = type(self).__name__ + '\n'
+        string += '{: <16}=\t{}\n'.format('\tEnergy', self.energy)
+        string += '{: <16}=\t{}\n'.format('\tInterpolant', self.y)
+        return string
+
+    @classmethod
+    def from_hdf5(cls, group, **kwargs):
```

Comment:
I did some magic with `FilterMeta`.  This should inherit the docstring from `Filter.from_hdf5`, even in Sphinx.